### PR TITLE
Support 'msf1' support in heic.c

### PR DESCRIPTION
**Pull request by liuchang-cc**
There are some heic file's ftyp valus is 'msf1', it's defined in heic-c066067_ISO_IEC_23008-12_2017.pdf

Analysis: {heif file's ftyp value is 'msf1' cause no decoder supported}
Solution: {add 'msf1' ftyp support in heic.c}

Signed-off-by: liuchang-cc <liuchang@oppo.com>

### Prerequisites

- [x] I have written a descriptive pull-request title
- [x] I have verified that there are no overlapping [pull-requests](https://github.com/ImageMagick/ImageMagick/pulls) open
- [x] I have verified that I am following the existing coding patterns and practices as demonstrated in the repository.

### CODE REVIEW

File: coders/gif.c [hunk 1]
```diff
@@ -1729,7 +1729,7 @@ static MagickBooleanType WriteGIFImage(const ImageInfo *image_info,Image *image,
         (void) WriteBlobByte(image,(unsigned char) (opacity >= 0 ? opacity :
           0));
         (void) WriteBlobByte(image,(unsigned char) 0x00);
-        if ((image->gamma != 1.0f/2.2f))
+        if (fabs(image->gamma - 1.0f/2.2f) > (1e-6))
```
Comment (dlemstra) [inline coders/gif.c:1732]: Can you please remove this from the PR? This is not related to the `msf1` changes.

File: coders/heic.c [hunk 2]
```diff
@@ -554,6 +554,7 @@ static Image *ReadHEICImage(const ImageInfo *image_info,
   file_data=RelinquishMagickMemory(file_data);
   if (status == MagickFalse)
     return(DestroyImageList(image));
+  (void) CloseBlob(image);
```
Comment (dlemstra) [inline coders/heic.c:557]: Thanks for this fix.

File: coders/heic.c [hunk 3]
```diff
@@ -1114,9 +1117,9 @@ static MagickBooleanType WriteHEICImage(const ImageInfo *image_info,
     */
     if (lossless != MagickFalse)
       error=heif_encoder_set_lossless(heif_encoder,1);
-    else if (image_info->quality != UndefinedCompressionQuality)
+    else if (image->quality != UndefinedCompressionQuality)
```
Comment (dlemstra) [inline coders/heic.c:1120]: Please revert these `image_info` to `image` changes. We don't want to use `image->quality` here.

File: coders/jpeg.c [hunk 4]
```diff
@@ -2436,7 +2436,7 @@ static MagickBooleanType WriteJPEGImage_(const ImageInfo *image_info,
 #endif
 #if (JPEG_LIB_VERSION >= 61) && defined(C_PROGRESSIVE_SUPPORTED)
   if ((LocaleCompare(image_info->magick,"PJPEG") == 0) ||
-      (image_info->interlace != NoInterlace))
+      (image->interlace != NoInterlace))
```
Comment (dlemstra) [inline coders/jpeg.c:2439]: Please revert.

File: coders/png.c [hunk 5]
```diff
@@ -10940,7 +10940,7 @@ static MagickBooleanType WriteOnePNGImage(MngInfo *mng_info,
     png_set_compression_strategy(ping,
        mng_info->write_png_compression_strategy-1);
 
-  ping_interlace_method=image_info->interlace != NoInterlace;
+  ping_interlace_method=image->interlace != NoInterlace;
```
Comment (dlemstra) [inline coders/png.c:10943]: Please revert changes in this file.

**Review by dlemstra — CHANGES_REQUESTED** [review]

File: coders/gif.c (hunk 1 above)
Comment (liuchang-cc) [reply coders/gif.c]: ok, I will create a new pull request for this commit. now I revert this commit firstly.

File: coders/heic.c (hunk 3 above)
Comment (liuchang-cc) [reply coders/heic.c]: ok, by the way, the quality value in image_info structure is zero now, did i missing something?

File: coders/png.c (hunk 5 above)
Comment (liuchang-cc) [reply coders/png.c]: done

File: coders/jpeg.c (hunk 4 above)
Comment (liuchang-cc) [reply coders/jpeg.c]: done

**Review by dlemstra — APPROVED** [review]
